lib/quoters/WebhookQuoter.ts

```diff
@@ -30,6 +32,19 @@ export class WebhookQuoter implements Quoter {
 
   public async quote(request: QuoteRequest): Promise<QuoteResponse[]> {
     const endpoints = await this.getEligibleEndpoints();
+    const addrToEndpointsMap = await this.complianceProvider.getAddrToEndpointsMap();
+    
+    endpoints.forEach((endpoint) => {
```

Comment:
could this instead just be a filter? splicing while iterating is a bit of a code smell